File: code/java/sdk/src/main/java/StarFederation/Datastar/events/MergeFragments.java
package StarFederation.Datastar.events;

import StarFederation.Datastar.enums.EventType;
import static StarFederation.Datastar.Consts.*;

import java.util.*;

public class MergeFragments extends AbstractSSEEvent {

    private final String data;
    private final MergeFragmentsOptions mergeFragmentsOptions;

    public MergeFragments(String data, MergeFragmentsOptions mergeFragmentsOptions) {
        super();
        if (data == null || data.isBlank()) {
            throw new IllegalArgumentException("Data cannot be null or empty.");
        }
        this.data = data.trim();
        this.mergeFragmentsOptions = Objects.requireNonNullElse(mergeFragmentsOptions, MergeFragmentsOptions.create());
    }

    @Override
    public EventType getEventType() {
        return EventType.MergeFragments;
    }

    public String[] getDataLines() {
        List<String> dataLines = new ArrayList<>();

        // Add Selector if Present
        addDataLineIfNotEmpty(dataLines, SELECTOR_DATALINE_LITERAL, mergeFragmentsOptions.getSelector());

        addDataLineIfNotDefault(
                dataLines, MERGE_MODE_DATALINE_LITERAL, mergeFragmentsOptions.getMergeMode().getValue(), DEFAULT_FRAGMENT_MERGE_MODE.getValue()
        );

        addDataLineIfNotDefault(
                dataLines, SETTLE_DURATION_DATALINE_LITERAL, mergeFragmentsOptions.getSettleDuration(), DEFAULT_SETTLE_DURATION
        );

        if (mergeFragmentsOptions.isUseViewTransition()) {
            dataLines.add(getDataLine(USE_VIEW_TRANSITION_DATALINE_LITERAL, getBooleanAsString(true)));
        }

        Arrays.stream(data.split("\n"))
                .map(String::trim)
                .filter(line -> !line.isEmpty())
                .forEach(line -> dataLines.add(getDataLine(FRAGMENTS_DATALINE_LITERAL, line)));

        return dataLines.toArray(String[]::new);
    }
}

File: sdk/php/src/events/MergeFragments.php
<?php
/**
 * @copyright Copyright (c) PutYourLightsOn
 */

namespace starfederation\datastar\events;

use Exception;
use starfederation\datastar\Consts;
use starfederation\datastar\enums\EventType;
use starfederation\datastar\enums\FragmentMergeMode;

class MergeFragments implements EventInterface
{
    use EventTrait;

    public string $fragments;
    public string $selector = '';
    public FragmentMergeMode $mergeMode = Consts::DEFAULT_FRAGMENT_MERGE_MODE;
    public int $settleDuration = Consts::DEFAULT_FRAGMENTS_SETTLE_DURATION;
    public bool $useViewTransition = Consts::DEFAULT_FRAGMENTS_USE_VIEW_TRANSITIONS;

    public function __construct(string $fragments, array $options = [])
    {
        $this->fragments = $fragments;

        foreach ($options as $key => $value) {
            if ($key === 'mergeMode') {
                $value = $this->getMergeMode($value);
            }

            $this->$key = $value;
        }
    }

    /**
     * @inerhitdoc
     */
    public function getEventType(): EventType
    {
        return EventType::MergeFragments;
    }

    /**
     * @inerhitdoc
     */
    public function getDataLines(): array
    {
        $dataLines = [];

        if (!empty($this->selector)) {
            $dataLines[] = $this->getDataLine(Consts::SELECTOR_DATALINE_LITERAL, $this->selector);
        }

        if ($this->mergeMode !== Consts::DEFAULT_FRAGMENT_MERGE_MODE) {
            $dataLines[] = $this->getDataLine(Consts::MERGE_MODE_DATALINE_LITERAL, $this->mergeMode->value);
        }

        if ($this->settleDuration != Consts::DEFAULT_FRAGMENTS_SETTLE_DURATION) {
            $dataLines[] = $this->getDataLine(Consts::SETTLE_DURATION_DATALINE_LITERAL, $this->settleDuration);
        }

        if ($this->useViewTransition !== Consts::DEFAULT_FRAGMENTS_USE_VIEW_TRANSITIONS) {
            $dataLines[] = $this->getDataLine(Consts::USE_VIEW_TRANSITION_DATALINE_LITERAL, $this->getBooleanAsString($this->useViewTransition));
        }

        return array_merge(
            $dataLines,
            $this->getMultiDataLines(Consts::FRAGMENTS_DATALINE_LITERAL, $this->fragments),
        );
    }

    private function getMergeMode(FragmentMergeMode|string $value): FragmentMergeMode
    {
        $value = is_string($value) ? FragmentMergeMode::tryFrom($value) : $value;

        if ($value === null) {
            $enumValues = array_map(fn($case) => '`' . $case->value . '`', FragmentMergeMode::cases());

            throw new Exception('An invalid value was passed into `mergeMode`. The value must be one of: ' . implode(', ', $enumValues) . '.');
        }

        return $value;
    }
}
